packages/ember-metal/lib/meta.js

```diff
@@ -326,7 +327,7 @@ export class Meta {
     let descriptor = lookupDescriptor(obj, key);
     let isMandatorySetter = descriptor && descriptor.set && descriptor.set.isMandatorySetter;
 
-    if (isMandatorySetter) {
+    if (isMandatorySetter === true) {
```

Comment:
I'm surprised it knows `l` is a boolean in `let foo = r && l` As `l` being boolean or not shouldn't change the value of `foo`, only `r` can do that.